### PR TITLE
Add new archives profile and several puppet module update

### PR DESCRIPTION
**Pull request by olblak**
The goal of this PR is to add a new role that could be used from the new arm machine running on oracle.
Because the ubuntu version must be 20.04 and it's arm architecture, several improvement needed to be done

1. Only install azure-cli on amd64 machines
2. Update apt puppet module
3. Update Letsencrypt  puppet module
4. Update firewall puppet module

To test this pull request, I run `vagrant up -d archives` for both ubuntu version 18.04 and 20.04

### CODE REVIEW

File: dist/role/manifests/archives.pp [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#
+# Okra is a tiny VM (1vCPU/4GB RAM) on Oracle
```
Comment (MarkEWaite) [inline dist/role/manifests/archives.pp]: I thought that we already had a machine named `okra`.  Should this have a distinct name?
```suggestion
# Okra is a tiny VM (1vCPU/4GB RAM) on Oracle
```

File: spec/server/archives/archives_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  it_behaves_like "a standard Linux machine"
+
+  describe service('apache2') do
+    it { should be_enabled }
+    it { should be_running }
+  end
+
+  describe port(80) do
+    it { should be_listening }
+  end
```
Comment (MarkEWaite) [inline spec/server/archives/archives_spec.rb:13]: Should port 443 be checked in the same way?

File: dist/role/manifests/archives.pp (hunk 1 above)
Comment (olblak) [reply dist/role/manifests/archives.pp]: Good point

File: dist/role/manifests/archives.pp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#
+# archives is a tiny VM (2vCPU/8GB RAM) on Oracle
```
Comment (dduportal) [inline dist/role/manifests/archives.pp]:
```suggestion
# archives requires low resources (2vCPU/8GB RAM)
```

File: dist/profile/manifests/azure.pp [hunk 4]
```diff
@@ -5,15 +5,11 @@
 class profile::azure (
   $cli = true,
 ) {
+  # azure-cli only works on amd64
```
Comment (dduportal) [inline dist/profile/manifests/azure.pp]:
```suggestion
```

File: dist/profile/manifests/azure.pp [hunk 5]
```diff
@@ -5,15 +5,11 @@
...
+  include apt
 
-  if $cli {
-    ensure_packages(['python-pip'])
+  if ($cli == true) and ($facts['architecture'] == 'amd64') {
```
Comment (dduportal) [inline dist/profile/manifests/azure.pp:11]:
```suggestion
  # azure-cli only works on amd64
  if ($cli == true) and ($facts['architecture'] == 'amd64') {
```

File: hieradata/common.yaml [hunk 6]
```diff
@@ -98,6 +98,10 @@ accounts:
 # Lets Encrypt settings
 letsencrypt::config::email: "tyler@monkeypox.org"
 letsencrypt::config::server: "https://acme-staging.api.letsencrypt.org/directory"
+letsencrypt::renew_cron_ensure: 'present'
+letsencrypt::renew_cron_minute: 0
+letsencrypt::renew_cron_hour: 6
+letsencrypt::renew_cron_monthday: '1-31/2'
```
Comment (dduportal) [inline hieradata/common.yaml]:
```suggestion
```

Comment (dduportal) [reply hieradata/common.yaml]: (default value is `*` which mean everyday)

**Pull request merged**